### PR TITLE
CGMES export - Add a parameter to export generators in local regulation mode

### PR DESCRIPTION
**Pull request by rcourtier**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

New feature

**What is the current behavior?**

Generator regulation in CGMES reflects the generator regulation in IIDM

**What is the new behavior (if this is a feature change)?**

It is now possible to set a parameter in the CGMES export to force the generator regulation mode to local: `iidm.export.cgmes.export-generators-in-local-regulation-mode`.
It only affects the CGMES generated files, not the IIDM model. Also it only concerns voltage regulating generators, not the reactive power regulating generators.
When the parameter is set to true:
- The Generator's RegulatigControl.Terminal in the CGMES EQ is set to the Generator's own terminal.
- The RegulatingControl.targetValue in the CGMES SSH is adapted (the engineering value is rescaled in accordance with the new terminal, but the per unit value remains the same)

When the parameter is set to false (default value), the old behaviour is kept: Generator regulation in CGMES reflects the generator regulation in IIDM.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No, but see the notice at the end of this description. Il should be reported in the release notes + in the migration guide.

**Other information**:

See https://github.com/powsybl/powsybl-core/pull/3192 for setting a generator's regulation mode to local in the IIDM model.

**Notice:**
The `targetV` on generator `"GEN"` is now set to `399 kV` instead of `24.5 kV` (to fix an inconsistency) for these methods of `EurostagTutorialExample1Factory`:
- `createWithRemoteVoltageGenerator()`
- `createWithRemoteReactiveAndVoltageGenerators()`

This may have an impact on other projects' unit tests.

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java [hunk 1]
```diff
@@ -393,14 +394,22 @@ private static void writeGenerators(Network network, Map<Terminal, String> mapTe
                             cimNamespace, writer, context);
                     break;
                 case CgmesNames.EXTERNAL_NETWORK_INJECTION:
-                    String regulatingControlId = RegulatingControlEq.writeRegulatingControlEq(generator, exportedTerminalId(mapTerminal2Id, regulatingTerminal), regulatingControlsWritten, mode, cimNamespace, writer, context);
+                    if (context.isExportGeneratorsInLocalRegulationMode()) {
+                        regulatingControlId = RegulatingControlEq.writeRegulatingControlEq(generator, exportedTerminalId(mapTerminal2Id, generator.getTerminal()), regulatingControlsWritten, mode, cimNamespace, writer, context);
+                    } else {
+                        regulatingControlId = RegulatingControlEq.writeRegulatingControlEq(generator, exportedTerminalId(mapTerminal2Id, regulatingTerminal), regulatingControlsWritten, mode, cimNamespace, writer, context);
+                    }
```
Comment (alicecaron) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java:401]: This code seems duplicated with the below case `case CgmesNames.SYNCHRONOUS_MACHINE:` maybe we can create a method to determine the `regulatingControlId` and use it in those 2 cases?

Comment (rcourtier) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java]: I let the original code but instead set above the proper value for the regulatingTerminal depending on the regulation mode.

Comment (alicecaron) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java]: Yes it seems good this way!

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java [hunk 2]
```diff
@@ -388,7 +388,14 @@ private static void addRegulatingControlView(Generator g, Map<String, List<Regul
                 targetValueUnitMultiplier = "M";
                 enabled = rrpc.isEnabled();
             } else {
-                target = g.getTargetV();
+                if (context.isExportGeneratorsInLocalRegulationMode()) {
```
[the remark this answers is not in the document]
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java:392]: done

[the remark this answers is not in the document]
Comment (olperr1) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java:392]: Can you just add a comment?
When reading the code, this test seems to be useless.

[the remark this answers is not in the document]
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java:392]: Done

**Review by olperr1 — APPROVED** [review]

**Review by alicecaron — APPROVED** [review]

**Pull request merged**